### PR TITLE
Computing a raised to b mod c in logn time

### LINKED ISSUE #101: Computing a raised to b mod c in logn time?
There is a fast algorithm to compute (a^b)%c in O(log N) time rather than O(b). Can I work on this issue?

### PR DESCRIPTION
**Pull request by yashagarwal1999**
Solved 
close #101

### CODE REVIEW

**Comment by rishabhgarg25699** [comment]
@yashagarwal1999 Give proper indentation to the code

**Comment by yashagarwal1999** [comment]
Added Indentation to the code

File: fast_power.cpp [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ll r = 1; 
+for (; k; k >>= 1)
+ {
+ if (k & 1) r = r * n%p; 
+ n = n * n%p;
+ } 
+ return r;
+ }
+ 
+
```
Comment (rishabhgarg25699) [inline fast_power.cpp:22]: No space required.

File: fast_power.cpp [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+int main(){
+    ios::sync_with_stdio(0);
+    cin.tie(0);cout.tie(0);
+     ll a,b,c;
+     cin>>a>>b>>c;
+     ll ans=fast_power(a,b,c);
+     cout<<ans<<endl;
+      return 0;
```
Comment (rishabhgarg25699) [inline fast_power.cpp:30]: Extra space

File: fast_power.cpp [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+#define MOD9 1000000009
+#define forr(i,p,n) for(ll i=p;i<n;i++)
+#define endl "\n"
+typedef long long ll;
+using namespace std;
+
+ll fast_power(ll n, ll k, ll p = MOD) {
+ll r = 1; 
+  for (; k; k >>= 1)
+   {
```
Comment (rishabhgarg25699) [inline fast_power.cpp:12]: Extra space

File: fast_power.cpp [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    if (k & 1)
```
Comment (rishabhgarg25699) [inline fast_power.cpp:13]: 4 line space after bracket.

File: fast_power.cpp [hunk 5]
```diff
@@ -0,0 +1,31 @@
+#include<bits/stdc++.h>
+#define MOD 1000000007
+#define MOD9 1000000009
+#define forr(i,p,n) for(ll i=p;i<n;i++)
+#define endl "\n"
+typedef long long ll;
+using namespace std;
+
+ll fast_power(ll n, ll k, ll p = MOD) {
+ll r = 1;
```
Comment (rishabhgarg25699) [inline fast_power.cpp:10]: No indentation.

**Review by rishabhgarg25699 — CHANGES_REQUESTED** [review]
@yashagarwal1999 Please make correct indentation.

**Pull request closed without merging**